editor/components/post-preview-button/index.js

```diff
@@ -123,14 +124,25 @@ export class PostPreviewButton extends Component {
 	}
 }
 
-export default connect(
-	( state ) => ( {
-		postId: state.currentPost.id,
-		link: getEditedPostPreviewLink( state ),
-		isDirty: isEditedPostDirty( state ),
-		isNew: isEditedPostNew( state ),
-		isSaveable: isEditedPostSaveable( state ),
-		modified: getEditedPostAttribute( state, 'modified' ),
+export default compose(
+	withSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( 'core/editor' );
+		const { getPostType } = select( 'core' );
+		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		return {
+			isPreviewable: postType && postType.previewable,
```

Comment:
You can use `get( postType, 'previewable', false )` here, rather than having the inline check.

Reply:
Actually, should probably just access it as `postType.previewable`, without the `postType &&`, as the posttype should always be retrieved in edit context.

Reply:
Updated.